### PR TITLE
[Merged by Bors] - feat: `BoundedSMul` for matrix norms

### PR DESCRIPTION
**Pull request by eric-wieser**
This provides the fact that the norm of scaling a matrix of quaternions by a quaternion is bounded by the product of the norms.

---

- [x] depends on: #9796

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Matrix.lean [hunk 1]
```diff
@@ -238,6 +245,13 @@ protected def linftyOpNormedAddCommGroup [NormedAddCommGroup α] :
   (by infer_instance : NormedAddCommGroup (m → PiLp 1 fun j : n => α))
 #align matrix.linfty_op_normed_add_comm_group Matrix.linftyOpNormedAddCommGroup
 
+
```
Comment (eric-wieser) [inline Mathlib/Analysis/Matrix.lean:248]:
```suggestion
```

**Comment by j-loreaux** [comment]
Can you add this for the operator norm (from `EuclideanSpace`) too? It's in `Analysis.NormedSpace.Star.Matrix`

**Comment by eric-wieser** [comment]
> Can you add this for the operator norm (from `EuclideanSpace`) too? It's in `Analysis.NormedSpace.Star.Matrix`

There's not really any point, because that uses `[IsROrC 𝕜]` and so there is no case where `BoundedSMul` is true and `NormedSpace` isn't.

**Comment by eric-wieser** [comment]
Ok, I guess it would be statable, but we're missing the corresponding instances on the operator norm on continuous linear maps:
```lean
theorem instL2OpNBoundedSmul [SeminormedRing R] [Module R 𝕜] [SMulCommClass 𝕜 R 𝕜]
    [BoundedSMul R 𝕜] : BoundedSMul R (Matrix m n 𝕜) :=
  .of_norm_smul_le fun r x => by
    rw [l2_op_norm_def]
    have := norm_smul_le r ((toEuclideanLin (𝕜 := 𝕜) (m := m) (n := n)).trans toContinuousLinearMap x) -- fails
    sorry
```

**Comment by j-loreaux** [comment]
Thanks. I'll add the bounded smul myself for continuous linear maps if I remember.

bors merge

**Comment by eric-wieser** [comment]
bors merge

(the failure was a caching issue)

**Pull request closed without merging**